### PR TITLE
Remote attendance

### LINKED ISSUE #1141: Add self-service attendance recording links
Short-lived links to say you were at a remote team meeting, similar to RSVPs with forced CAS auth

### PR DESCRIPTION
**Pull request by kberzinch**
Closes #1141

### CODE REVIEW

File: app/Nova/Actions/ResetRemoteAttendance.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     * Get the fields available on the action.
+     *
+     * @return array<\Laravel\Nova\Fields\Field>
+     */
+    public function fields(): array
+    {
+        if (Auth::user()->hasRole('admin')) {
+            return [
+                DateTime::make('Expiration Time', 'expiration_time')
+                    ->required(true)
```
Comment (JoshOldenburg) [inline app/Nova/Actions/ResetRemoteAttendance.php]: I'd suggest making this optional and saying in the help text that it defaults to 1 hour in the future.

File: app/Nova/Team.php [hunk 2]
```diff
@@ -159,6 +161,39 @@ protected function controlFields(): array
...
+    {
+        return [
+            Text::make('Link', 'attendance_secret')
+                ->hideFromIndex()
+                ->resolveUsing(static function (?string $secret): ?string {
+                    return null === $secret ? null : config('app.url').'attendance/remote/'.$secret;
+                })
+                ->readonly(static function (Request $request): bool {
+                    return ! $request->user()->hasRole('admin');
+                })
```
Comment (JoshOldenburg) [inline app/Nova/Team.php]: I don't think this should be editable at all. If it is going to be editable, there should be a separate Nova field for the secret itself with no custom resolve function so you don't end up with the attendance_secret having the `https://apiary-test.robojackets.org/attendance/remote/` prefix in it (which makes the link resolve as `https://apiary-test.robojackets.org/attendance/remote/https://apiary-test.robojackets.org/attendance/remote/randomcharactershere` if you don't change it).

Comment (kberzinch) [reply app/Nova/Team.php]: In the interest of keeping admin superpowers 🙂 I'd like to keep it editable, though I will make a separate field as you suggest.

File: app/Nova/Actions/ResetRemoteAttendance.php (hunk 1 above)
Comment (kberzinch) [reply app/Nova/Actions/ResetRemoteAttendance.php]: Heh, despite the red asterisk nova will let you run the action without specifying a timestamp and it does work as intended.

**Pull request merged**